src/components/Lesson/Lesson.js

```diff
@@ -91,88 +37,44 @@ const rememberLastLesson = (path, setLastLesson) => {
   setLastLesson(lessonPath);
 };
 
-const Lesson = React.createClass({
-  getTitle() {
-    return this.props.lesson.frontmatter.title || this.props.params.file;
-  },
-  getLevel() {
-    return this.props.lesson.frontmatter.level || 0;
-  },
-  getAuthor() {
-    const author = this.props.lesson.frontmatter.author || '';
-    return author ?
-      <p><i>{this.props.t('lessons.writtenby')} <MarkdownRenderer src={author} inline={true} /></i></p> :
-      null;
-  },
-  getTranslator() {
-    const translator = this.props.lesson.frontmatter.translator || '';
-    return translator ?
-      <p><i>{this.props.t('lessons.translatedby')} <MarkdownRenderer src={translator} inline={true} /></i></p> :
-      null;
-  },
-  getLanguage() {
-    return this.props.lesson.frontmatter.language || '';
-  },
-  createMarkup(){
-    return {
-      __html: removeHtmlFileEnding(this.props.lesson.content)
-    };
-  },
-  setLanguage(){
-    const lessonLanguage = this.getLanguage();
-    if(lessonLanguage !== '' && lessonLanguage !== this.props.language) {
-      this.props.setLanguage(lessonLanguage);
-    }
-  },
-  componentWillMount(){
-    if (typeof document === 'undefined') {
-      // do nothing server-side
-      return;
-    }
-    this.props.lesson.content = processContent(this.props.lesson.content, contentStyles);
-
-    if(this.props.isReadme) this.props.setModeTeacher();
+const createMarkup = (lessonContent) => {
+  if (typeof document !== 'undefined') {
+    return ({__html: removeHtmlFileEnding(processContent(lessonContent, contentStyles))});
+  }
+  return;
+};
 
-    //Changes the language state to the language defined in the current lesson or readme-file
-    //this.setLanguage();
-  },
+const Lesson = React.createClass({
   componentDidMount() {
     const {path, checkboxes, setCheckbox, setLastLesson} = this.props;
     setCheckboxes(path, checkboxes, setCheckbox);
     rememberLastLesson(path, setLastLesson);
     renderToggleButtons();
   },
-  componentWillUnmount() {
-    const nodes = [...document.getElementsByClassName('togglebutton')];
-    for (let node of nodes) {
-      ReactDOM.unmountComponentAtNode(node);
-    }
-  },
   render() {
-    const {t, path, lessons, isReadme, isStudentMode, setCheckbox, checkboxes} = this.props;
-    const instructionBtn = isReadme ? <LessonButton {...{path, lessons, t}}/> :
-      isStudentMode ? null : <ReadmeButton {...{path, lessons, t}}/>;
-    const resetButton = anyCheckboxTrue(checkboxes) === true ?
-      <Button className={styles.resetButton}  bsStyle="warning" bsSize="small"
-      onClick={() => setCheckboxes(path, {}, setCheckbox)}>{t('lessons.reset')}</Button>
-      : null;
+    const {t, path, checkboxes, params, lesson, title, level, authorName,
```

Comment:
Bare for å gjøre det litt mer oversiktlig, kunne du delt inn i tre linjer, slik at man ser hva som kommer fra ownProps, mapStateToProps, og mapDispatchToProps? F.eks.
```
const {
  path, params, ...,
  t, checkboxes, ...,
  setCheckbox, ...,
} = this.props;
```